### ISSUE TITLE
Handling more than 4 LADSPA effects via the GUI

### ISSUE THREAD

**Issue by theGreatWhiteShark** [hydrogen-music/hydrogen#1151]
In the `CMakeLists.txt` file the number of LADSPA effects used by Hydrogen can be set. While the core part seems to support and arbitrary amount, the GUI is designed to only handle up to four of them.

![FX_mixer](https://user-images.githubusercontent.com/14164547/106387744-7103e380-63db-11eb-9d58-323054fd4005.png)

Notice that the control of the FX to the right is not cut by my display/screenshot but by Hydrogen itself.

**Comment by oddtime** [comment]
If some is willing and able to change the mixer (in order to make more sends possible) I would appreciate some other features as well, but it is takes a lot of time.
I proposed some valuetips in #1013 to show the hidden values (faders, knobs...).

Does the core support FX inserts?

**Comment by theGreatWhiteShark** [comment]
> I proposed some valuetips in #1013 to show the hidden values (faders, knobs...).

Right. Haven't looked into it yet.

> Does the core support FX inserts?

Not sure what you mean. I haven't tried it but I would suspect the core can handle and arbitrary number of FX (since it loops over them till `MAX_FX`. But you can not increase the number after compiling Hydrogen. But I'm not sure whether this is due to some technical limitations or just a remnant and could be converted into an option. Anyway, it's probably just the GUI that needs to be fixed.

> If somebody is willing and able

Well, at least at my end plans for enhancements/new features already piled to such heights it will take a at least 1-2 years at my current pace. But if you are interested, you can give it a shot :wink:

**Comment by oddtime** [comment]
> Not sure what you mean

You are right, that's a stupid question, of course not now but if mixer can handle an effect send, the core could be modified to handle chain effects as inserts.

> Well, at least at my end plans for enhancements/new features already piled to such heights it will take a at least 1-2 years at my current pace. But if you are interested, you can give it a shot

Yes I imagine, no problem! I didn't mean to delegate this. I don't have skills to redesign the GUI, and have some other priorities for H2 when I have time ;)

**Comment by theGreatWhiteShark** [comment]
> but if mixer can handle an effect send, the core could be modified to handle chain effects as inserts.

It's not a true send but rather a flag to tell the core to apply a couple of LADSPA effects on the audio before handing them to the output ports. There is no audio routing in Hydrogen itself.

Haven't tried it but AFAIK `JACK` is available on all platforms Hydrogen is supported on. Using its routing + dedicated mixer applications and/or plugin hosts you can build up a much more flexible and scalable (since Hydrogen is using just one core for processing all it's audio) setup.
